tests/e2e/test_spyre_cb.py

```diff
@@ -87,149 +86,134 @@ def test_cb_handling(
-        assert len(engine_core.scheduler.running) == 1
-
-        # add another request
-        engine.add_request("2", prompt2, sampling_params)
-
-        assert len(engine_core.scheduler.waiting) == 1  # req 2 is waiting
-        assert len(engine_core.scheduler.running) == 1
-
-        request_outputs = engine.step()
-        assert len(request_outputs) == 1  # still only 1 request
-        assert request_outputs[0].request_id == "2"  # req 2 is decoding
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 2
-
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2
-        assert request_outputs[0].request_id == "2"  # req 2 is decoding
-        assert request_outputs[1].request_id == "1"  # req 1 is decoding
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 2
-
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2
-        assert request_outputs[0].request_id == "2"  # req 2 is decoding
-        assert request_outputs[1].request_id == "1"  # req 1 is decoding
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 2
-
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2
-        assert request_outputs[0].request_id == "2"  # req 2 is decoding
-        assert request_outputs[1].request_id == "1"  # req 1 is decoding
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 2
-
-        # add the third request
-        # but since VLLM_SPYRE_MAX_BATCH_SIZE=2
-        # this request is waiting
-        engine.add_request("3", prompt3, sampling_params)
-
-        assert len(engine_core.scheduler.waiting) == 1
-        assert len(engine_core.scheduler.running) == 2
-
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2  # both requests decoding now
-        assert request_outputs[0].request_id == "2"  # req 2 is decoding
-        assert request_outputs[1].request_id == "1"  # req 1 is decoding
-
-        assert len(engine_core.scheduler.waiting) == 1
-        assert len(engine_core.scheduler.running) == 2
-
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2  # both requests decoding now
-        assert request_outputs[0].request_id == "2"  # req 2 is decoding
-        assert request_outputs[1].request_id == "1"  # req 1 is decoding
-        assert request_outputs[1].finished  # request 1 is done
-        assert request_outputs[1].outputs[0].text == " 3 2 "
 
-        assert len(engine_core.scheduler.waiting) == 1
-        assert len(engine_core.scheduler.running) == 1
+    # set env vars
+    monkeypatch.setenv("VLLM_SPYRE_USE_CB", "1")
+    monkeypatch.setenv("VLLM_USE_V1", "1")
+    monkeypatch.setenv("VLLM_SPYRE_DYNAMO_BACKEND", backend)
```

Comment:
Interesting, I don't really have a strong opinion on how `monkeypatch` should be used. We have it used as is and with `context` in spyre so doesn't really break convention. IMO I like the `context` use just because I know visually that once the context scope is over the env settings will definitely be removed.

Reply:
I guess it is really about personal preference here. I personally don't like to have extra indentation if not strictly unnecessary, especially with the formatter enforcing line length of 80. When using the monkeypath context here, and with my extra comment "(prefill step)", the corrected output of the formatter is a bit ugly, with lines that get wrapped. If that's okay we can remove the context here and I will change it everywhere else in a separate PR as you suggested?

Reply:
Oh interesting! I did not think about that. Yeah sure that works!